clients/drcachesim/tools/invariant_checker.h

```diff
@@ -38,13 +38,41 @@
 
 #include "analysis_tool.h"
 #include "dr_api.h"
+#include <iostream>
 #include "memref.h"
 #include <memory>
 #include <mutex>
 #include <stack>
 #include <unordered_map>
 #include <vector>
 
+/* The auto cleanup wrapper of instr_t.
+ * This can ensure the instance of instr_t is cleaned up when it is out of scope.
+ */
+struct instr_autoclean_t {
+public:
+    instr_autoclean_t(void *drcontext)
+        : drcontext(drcontext)
+    {
+        data = instr_create(drcontext);
+    }
+    ~instr_autoclean_t()
+    {
+#ifdef DEBUG
```

Comment:
This doesn't need to be a debug-only check. Also it'll be more useful in the constructor rather than here.

Reply:
Done.